simulator/src/main/java/peersim/kademlia/das/DASDHTProtocolNonValidator.java

```diff
@@ -0,0 +1,240 @@
+package peersim.kademlia.das;
+
+import java.math.BigInteger;
+import peersim.core.CommonState;
+import peersim.kademlia.KademliaObserver;
+import peersim.kademlia.Message;
+import peersim.kademlia.das.operations.RandomSamplingOperationDHT;
+import peersim.kademlia.das.operations.SamplingOperation;
+import peersim.kademlia.das.operations.ValidatorSamplingOperationDHT;
+import peersim.kademlia.operations.GetOperation;
+import peersim.kademlia.operations.Operation;
+
+public class DASDHTProtocolNonValidator extends DASDHTProtocol {
+
+  protected static String prefix = null;
+
+  public DASDHTProtocolNonValidator(String prefix) {
+    super(prefix);
+    DASDHTProtocolNonValidator.prefix = prefix;
+    isValidator = false;
+    isBuilder = false;
+  }
+
+  public Object clone() {
+    DASDHTProtocolNonValidator dolly =
+        new DASDHTProtocolNonValidator(DASDHTProtocolNonValidator.prefix);
+    return dolly;
+  }
+
+  /**
+   * Start a topic query opearation.<br>
+   *
+   * @param m Message received (contains the node to find)
+   * @param myPid the sender Pid
+   */
+  protected void handleInitNewBlock(Message m, int myPid) {
+    super.handleInitNewBlock(m, myPid);
+    logger.warning("non-validator new block:" + currentBlock.getBlockId());
+    startRandomSampling();
+  }
+
+  @Override
+  public void putValueReceived(Object o) {
+    logger.warning("Non-validator Parcel received put operation ");
+  }
+
+  protected boolean doSampling(SamplingOperation sop) {
+
+    if (sop.completed()) {
+      samplingOp.remove(sop.getId());
+      KademliaObserver.reportOperation(sop);
+      logger.warning(
+          "Sampling operation finished dosampling "
+              + sop.getId()
+              + " "
+              + CommonState.getTime()
+              + " "
+              + sop.getTimestamp());
+
+      return true;
+    } else {
```

Comment:
I've read somewhere that we should avoid "else" if the the first "if" has a return. It's a small thing but remove one level of indent :)

Reply:
solved